config/v1/types_ingress.go

```diff
@@ -82,8 +82,52 @@ type IngressSpec struct {
 	// Note that if there are no RequiredHSTSPolicies, any HSTS Policy annotation on the route is valid.
 	// +optional
 	RequiredHSTSPolicies []RequiredHSTSPolicy `json:"requiredHSTSPolicies,omitempty"`
+
+	// platform holds desired information specific to the underlying
+	// infrastructure provider for the ingress.
+	Platform *IngressPlatformSpec `json:"platform,omitempty"`
+}
+
+type IngressPlatformSpec struct {
+	// type is the underlying infrastructure provider for the cluster.
+	// Allowed values are "AWS", "Azure", "BareMetal", "GCP", "Libvirt",
+	// "OpenStack", "VSphere", "oVirt", "KubeVirt", "EquinixMetal", "PowerVS",
+	// "AlibabaCloud", "Nutanix" and "None". Individual components may not support all platforms,
+	// and must handle unrecognized platforms as None if they do not support that platform.
+	//
+	// +unionDiscriminator
+	Type PlatformType `json:"type"`
+
+	// AWS contains settings specific to the Amazon Web Services infrastructure provider.
+	// +optional
+	AWS *AWSIngressSpec `json:"aws,omitempty"`
+}
+
+type AWSIngressSpec struct {
+	// LoadBalancer contains the AWS load balancer details
+	LoadBalancer *LoadBalancer `json:"loadbalancer,omitempty"`
 }
 
+type LoadBalancer struct {
+	// LBType allows user to set a load balancer type.
+	// When this field is set the default ingresscontroller will get created using the specified LBType.
+	// If this field is not set then the default ingress controller of LBType Classic will be created.
+	// Allowed values are NLB or Classic
+	// +kubebuilder:validation:Enum:=NLB;Classic;""
+	// +optional
+	DefaultLBType AWSLBType `json:"defaultLBType,omitempty"`
```

Comment:
Lets make this `DefaultType`, we already know its to do with load balancers because it's in the load balancers section

The comment on this field also needs to be updated